### ISSUE TITLE
Identify parts of the models that are set in '.eval()' mode during initialization and keep them in eval mode after validation epoch ends

### ISSUE THREAD

**Issue by ductai199x** [Lightning-AI/lightning#18930]
### Description & Motivation

Currently, if I want a submodule of my model completely **frozen** during training, I cannot do so because even with 

1. Setting this submodule in `.eval()` mode in `__init__()`
2. Setting all `required_grad` of all parameters in this submodule to `False`
3. Using `torch.no_grad()` context when passing data through this module

If my submodule contains ANY batch-normalization layers, these will not stay in `.eval()` mode because the whole model (which contains this submodule) is set to `.train()` mode after validation epoch ends. Hence, these batch-normalization layers' running stats will change and affect the performance and training of the entire model.

This behavior is very easy to create a minimal example to see this effect.

### Pitch

Search for params that are set in eval mode during initialization and keep them in eval mode after validation epoch ends.

cc @borda
